### PR TITLE
Fork from ReBus.MongoDb 0.84 with upgraded MongoDb Driver to 2.7.2

### PR DESCRIPTION
**Pull request by dbuenor**
Due to the fact that we still using ReBus0.84 version, we need to upgrade ReBus.MongoDb library to allow sagas be created on a recent version for Mongo (>4.0), so that's why we have upgraded and updated ReBus.MongoDb to the latest MongoDb.Driver version.

We also added MongoDb tests to the project to check normal behaviour for the library. 

_PS: It needs a local mongodb server to pass tests._

### CODE REVIEW

File: Rebus.MongoDb/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (pruiz) [inline Rebus.MongoDb/Properties/AssemblyInfo.cs]: We need to automate build/deployment. As a start, a project at appveyor should be able to build this automatically (including setting a version to each build).

We should probably use 0.84.0.$BUILDNO$ as AssemblyFileVersion, while keeping a fixed 0.84.0 as AssemblyVersion.

File: Rebus.MongoDb/ReBus.MongoDb2.nuspec [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<package>
+	<metadata>
+		<id>Rebus.MongoDb2</id>
+		<version>0.0.0.0</version>
```
Comment (pruiz) [inline Rebus.MongoDb/ReBus.MongoDb2.nuspec]: This version should be handled too..

File: Rebus.MongoDb/Rebus.MongoDb.csproj [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+    <DocumentationFile>bin\Release\Rebus.MongoDb.XML</DocumentationFile>
+    <Prefer32Bit>false</Prefer32Bit>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="DnsClient, Version=1.0.7.0, Culture=neutral, PublicKeyToken=4574bb5573c51424, processorArchitecture=MSIL">
+      <HintPath>..\packages\DnsClient.1.0.7\lib\net45\DnsClient.dll</HintPath>
```
Comment (pruiz) [inline Rebus.MongoDb/Rebus.MongoDb.csproj]: What do we need DnsClient for?

File: Rebus.Tests/Integration/Factories/MsmqBusFactory.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+using Rebus.Shared;
```
Comment (pruiz) [inline Rebus.Tests/Integration/Factories/MsmqBusFactory.cs]: Not sure if this is needed, if unneeded, remove unrelated test files in order to reduce clutter.

File: Rebus.Tests/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (pruiz) [inline Rebus.Tests/Properties/AssemblyInfo.cs:34]: We are not releasing this assembly, but we may take care of this version strings too.

File: Rebus.Tests/packages.config [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Autofac" version="2.6.3.862" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: Rebus.Tests/packages.config [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="Castle.Core" version="3.2.0" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: Rebus.Tests/packages.config [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="Castle.Windsor" version="3.2.0" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: Rebus.Tests/packages.config [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="CommonServiceLocator" version="1.0" targetFramework="net45" />
+  <package id="DnsClient" version="1.0.7" targetFramework="net45" />
+  <package id="DryIoc.dll" version="2.0.0-preview181" targetFramework="net45" />
+  <package id="log4net" version="2.0.1" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config:4]: Do we need this?

File: Rebus.Tests/packages.config [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="MongoDB.Bson" version="2.7.2" targetFramework="net45" />
+  <package id="MongoDB.Driver" version="2.7.2" targetFramework="net45" />
+  <package id="MongoDB.Driver.Core" version="2.7.2" targetFramework="net45" />
+  <package id="NUnit" version="3.11.0" targetFramework="net45" />
+  <package id="RabbitMQ.Client" version="3.4.0" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: Rebus.Tests/packages.config [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="Rebus" version="0.84.0" targetFramework="net45" />
+  <package id="Rebus.Castle.Windsor" version="0.84.0" targetFramework="net45" />
+  <package id="Rebus.RabbitMQ" version="0.84.0" targetFramework="net45" />
+  <package id="RhinoMocks" version="3.6" targetFramework="net45" />
+  <package id="Serilog" version="1.4.118" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: Rebus.Tests/packages.config [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="Shouldly" version="2.1.1" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config:12]: Do we need this?

File: Rebus.Tests/packages.config [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="SimpleInjector" version="2.5.0" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: Rebus.Tests/packages.config [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  <package id="structuremap" version="3.0.3.116" targetFramework="net45" />
```
Comment (pruiz) [inline Rebus.Tests/packages.config]: Do we need this?

File: lib/Ponder/Ponder.cs [hunk 15]
```diff
@@ -0,0 +1,70 @@
+using System;
```
Comment (pruiz) [inline lib/Ponder/Ponder.cs]: A source code file within lib folder?

File: Rebus.MongoDb/Rebus.MongoDb.csproj (hunk 3 above)
Comment (dbuenor) [reply Rebus.MongoDb/Rebus.MongoDb.csproj]: It's a MongoDb.Driver.Core dependency.

File: Rebus.Tests/Integration/Factories/MsmqBusFactory.cs (hunk 4 above)
Comment (dbuenor) [reply Rebus.Tests/Integration/Factories/MsmqBusFactory.cs]: I did a commit to cleanup all unused stuff from tests. Removing this file too.

File: lib/Ponder/Ponder.cs (hunk 15 above)
Comment (dbuenor) [reply lib/Ponder/Ponder.cs]: This file originally was only on lib folder on Rebus project. I added to the Rebus.MongoDb project to avoid having file outside of the folder project, but forget to remove this duplicated file. So, i remove this file and let only Ponder.cs which is in the ReBus.MongoDb project folder.

File: Rebus.Tests/packages.config (hunk 6 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 7 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 8 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 9 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config:4]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 10 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 11 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 13 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 14 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config]: Removed all unused nugets from project.

File: Rebus.Tests/packages.config (hunk 12 above)
Comment (dbuenor) [reply Rebus.Tests/packages.config:12]: Removed all unused nugets from project.

File: Rebus.MongoDb/ReBus.MongoDb2.nuspec (hunk 2 above)
Comment (dbuenor) [reply Rebus.MongoDb/ReBus.MongoDb2.nuspec]: Version changed to $version$ and added missing dependencies too (ReBus and MongoDb)
